src/Authoring/WinRT.SourceGenerator/DiagnosticRules.cs

```diff
@@ -0,0 +1,150 @@
+﻿using Microsoft.CodeAnalysis;
+
+namespace WinRT.SourceGenerator
+{
+    public class DiagnosticRules
+    {
+        /// <summary>Helper function that does most of the boilerplate information needed for Diagnostics</summary> 
+        /// <param name="id">string, a short identifier for the diagnostic </param>
+        /// <param name="title">string, a few words generally describing the diagnostic</param>
+        /// <param name="messageFormat">string, describes the diagnostic -- formatted with {0}, ... -- 
+        /// such that data can be passed in for the code the diagnostic is reported for</param>
+        private static DiagnosticDescriptor MakeRule(string id, string title, string messageFormat)
+        {
+            return new DiagnosticDescriptor(
+                id: id,
+                title: title,
+                messageFormat: messageFormat,
+                category: "Usage",
+                /* Warnings dont fail command line build; winmd generation is prevented regardless of severity.

```

Comment:
Did you plan on addressing this or future PR?

Reply:
Yes we want to do it now, thanks

Reply:
addressed